### PR TITLE
[PROTOCOL-440] Merge EtherCollateralLoans and TokenCollateralLoans into LoansBase

### PR DESCRIPTION
**Pull request by sypeer**
It is:
- Merging EtherCollateralLoans & TokenCollateralLoans into LoansBase

### CODE REVIEW

File: contracts/base/LoansBase.sol [hunk 1]
```diff
@@ -622,4 +644,22 @@ contract LoansBase is LoansInterface, Base {
...
+    }
+
+    /**
+        @notice It transfers an amount of collateral tokens from an address to this contract.
+        @param from The address where the tokens will transfer from.
+        @param amount The amount to be transferred.
+     */
+    function _collateralTokenTransferFrom(address from, uint256 amount) internal {
+        ERC20Detailed(collateralToken).safeTransferFrom(from, address(this), amount);
+    }
```
Comment (passabilities) [inline contracts/base/LoansBase.sol:664]: we can inline these functions

File: contracts/base/LoansBase.sol [hunk 2]
```diff
@@ -582,6 +595,15 @@ contract LoansBase is LoansInterface, Base {
         @param amount The amount of collateral to be paid
      */
     function _payInCollateral(uint256 loanID, uint256 amount) internal {
+        if (collateralToken == _getSettings().ETH_ADDRESS()) {
+            // Ether collateral
+            require(msg.value == amount, "INCORRECT_ETH_AMOUNT");
```
Comment (passabilities) [inline contracts/base/LoansBase.sol:600]: we should change the collateral token to be the `WETH` contract address that is on settings

also here is the point in which we should wrap the ETH value in WETH. after this point the only other point the contract needs to know if the collateral token is ETH or not is upon withdrawing. then we should be unwrapping the WETH and sending the caller back their unwrapped ETH collateral.

File: contracts/base/LoansBase.sol [hunk 3]
```diff
@@ -554,7 +554,17 @@ contract LoansBase is LoansInterface, Base {
         uint256 loanID,
         uint256 amount,
         address payable recipient
-    ) internal;
+    ) internal {
+        totalCollateral = totalCollateral.sub(amount);
+        loans[loanID].collateral = loans[loanID].collateral.sub(amount);
+        if (collateralToken == _getSettings().ETH_ADDRESS()) {
+            // Ether collateral
+            recipient.transfer(amount);
```
Comment (passabilities) [inline contracts/base/LoansBase.sol:562]: here we we unwrap the WETH and then transfer

**Review by passabilities — CHANGES_REQUESTED** [review]

File: contracts/base/LoansBase.sol (hunk 2 above)
Comment (sypeer) [reply contracts/base/LoansBase.sol:602]: Done

File: contracts/base/LoansBase.sol (hunk 1 above)
Comment (sypeer) [reply contracts/base/LoansBase.sol]: Moved

File: contracts/base/LoansBase.sol (hunk 3 above)
Comment (sypeer) [reply contracts/base/LoansBase.sol]: done

**Pull request closed without merging**